### ISSUE TITLE
Question: userId

### ISSUE THREAD

**Issue by hrdwdmrbl** [thenickdude/PlanC#25]
When you have a user-defined password, do you still need to provide the userId?
Basically I didn't end up needing it.
I am pretty sure that I did even found my userId, but when I tried using it then I couldn't decrypt.
I also tried cracking the userId but after 24h nothing had been found.

So yeah, what's the deal with userId? Is it sometimes not needed? Could you explain because the documents seem to say that its always needed.

**Comment by thenickdude** [comment]
It's needed whenever my app prompts you to enter it. 

The user ID is used as a salt in deriving the encryption key from a password, so without the userid the encryption key can't be recovered.

Are you using Crashplan Home or is this Small Business?

**Comment by hrdwdmrbl** [comment]
Home... I really never entered it and I was able to decrypt... did the software find the key on its own? I'm so confused

**Comment by thenickdude** [comment]
Which command did you use to recover the encryption key exactly? Maybe your encryption key was not derived from a password at all?

**Comment by hrdwdmrbl** [comment]
I recovered like this:

`./plan-c --cpproperties /Volumes/XXX/Crashplan\ Archives/YYY/cp.properties --archive /Volumes/XXX/Crashplan\ Archives/YYY restore --dest /Volumes/XXX/Crashplan\ Restores\ 2/YYY`

**Comment by thenickdude** [comment]
Yeah seems like you didn't have a key derived from a password so PlanC was able to just read the encryption key from the cp.properties file directly.

**Comment by hrdwdmrbl** [comment]
How can I verify and confirm that that’s the case?
